### PR TITLE
chore(upgrade): v1.6.0 to v1.7.0

### PR DESCRIPTION
**Pull request by enddynayn**
- Upgrade Polkadot-sdk to v.1.7.0.
- Update weights to reflect the new version.

Notable Changes:
- [Allow custom error types in Jsonrpsee](https://github.com/paritytech/polkadot-sdk/pull/1313/)

For more details, please refer to:

[Release Notes](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.7.0)

#1870

### CODE REVIEW

File: node/service/src/service.rs [hunk 1]
```diff
@@ -505,6 +505,7 @@ fn start_consensus(
 		proposer,
 		collator_service,
 		authoring_duration: Duration::from_millis(1500),
+		reinitialize: true, // look into this one
```
Comment (enddynayn) [inline node/service/src/service.rs:508]: looking into it.

File: common/helpers/src/rpc.rs [hunk 2]
```diff
@@ -1,19 +1,19 @@
 use core::result::Result as CoreResult;
 use jsonrpsee::{
-	core::{Error as RpcError, RpcResult},
```
Comment (claireolmstead) [inline common/helpers/src/rpc.rs:3]: love removing an `as`

**Review by claireolmstead — APPROVED** [review]
Happy to approve, but I don't feel like I'm qualified enough to understand the purpose of a lot of these changes, specifically in reference to the `parts` value changes.

Lots of good type changes.

**Review by aramikm — APPROVED** [review]
Looks great!

 I think there might be some questions about how this back-pressure implementation which supports 65 messages per connection by default might have some effects on our apps subscribing to the chain.

File: common/helpers/src/rpc.rs (hunk 2 above)
Comment (saraswatpuneet) [reply common/helpers/src/rpc.rs:3]: 😂

**Review by saraswatpuneet — APPROVED** [review]
Read through code and comments

Great job 👏🏽

File: pallets/schemas/src/tests/serde_tests.rs [hunk 3]
```diff
@@ -46,7 +46,7 @@ fn serde_helper_deserialzer_error() {
 			r#"{ 56: "number" }"#,                    // KeyMustBeAString
 			r#"{ "file address": "file path" \r\n}"#, // EofWhileParsingObject
 			r#"{ "unicode code point": "\ud83f" }"#,  // InvalidUnicodeCodePoint
-			r#"{ "v": 300e715100 }"#,                 // NumberOutOfRange
+			                                          // r#"{ "v": 300e715100 }"#,              // NumberOutOfRange
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply pallets/schemas/src/tests/serde_tests.rs:49]: nit: did we find what is the underlying reason that this case don't fail no more ?

[the remark this answers is not in the document]
Comment (enddynayn) [reply pallets/schemas/src/tests/serde_tests.rs:49]: @aramikm and I checked it out but couldn’t find a solid answer. We also looked into any possible effects on the project and didn’t find any.

@saraswatpuneet Thanks for also taking a look into it and helping me find the issue.

**Comment by enddynayn** [comment]
> Looks great!
> 
> I think there might be some questions about how this back-pressure implementation which supports 65 messages per connection by default might have some effects on our apps subscribing to the chain.

Thats a good point. Ill create an issue so that we can properly configure backpressure.

**Pull request merged**